### PR TITLE
#169258568 setup testing environment

### PR DESCRIPTION
**Pull request by hezronkimutai**
#### What does this PR do?
This PR sets up a testing environment
#### Description of Task to be completed?
- Write jest configurations
- write one test that successfully passes
#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend ` 
-  `git checkout ch-setup-testing-169258568`
- `npm i`
- `npm test`
#### Any background context you want to provide?
CircleCI is expected to fail because I don't have .circleCi file
#### What are the relevant pivotal tracker stories?
[#169258568](https://www.pivotaltracker.com/n/projects/2407428/stories/169258568)
#### Screenshots (if appropriate)
Screen Shot 2019-12-24 at 19 16 30

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  moduleNameMapper: {
+    '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
+     '<rootDir>/src/testConfigurations/__mocks__/fileMock.js',
+    '\\.(css|scss)$': 'identity-obj-proxy',
+  },
+  setupFiles: [
+    '<rootDir>/src/testConfigurations/setupTests.js',
+  ],
+  testEnvironment: 'node',
+};
```
Comment (william000000) [inline jest.config.js]: Nice Shot! @hezronkimutai

**Review by william000000 — CHANGES_REQUESTED** [review]
Good Job @hezronkimutai, however you may put coverage folder in .gitignore file

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  coverageDirectory: 'coverage',
+  moduleNameMapper: {
+    '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
+     '<rootDir>/src/testConfigurations/__mocks__/fileMock.js',
+    '\\.(css|scss)$': 'identity-obj-proxy',
+  },
+  setupFiles: [
+    '<rootDir>/src/testConfigurations/setupTests.js',
+  ],
+  testEnvironment: 'node',
```
Comment (NiyoEric) [inline jest.config.js]: We are doing frontend so `testEnvironment: 'jsdom'`.

File: package.json [hunk 3]
```diff
@@ -4,8 +4,9 @@
   "description": "Develop Barefoot nomad frontend",
   "main": "index.js",
   "scripts": {
-    "test": "jest",
-    "dev": "webpack-dev-server --mode development --hot --open --config webpack.config.babel.js",
+    "coveralls": "jest --coverage",
+    "test": "jest && jest --coverage",
+    "dev": "webpack-dev-server --mode development --hot --open --config webpack.config.js",
```
Comment (NiyoEric) [inline package.json]: Webpack is configured in ES6. Therefore you need `@babel/register` to transpile. Replace script dev command with `"dev": "webpack-dev-server --mode development --hot --open --config webpack.config.babel.js",`

File: public/index.html [hunk 4]
```diff
@@ -5,6 +5,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>Barefoot Nomad</title>
+    <title>Crime-App</title>
```
Comment (NiyoEric) [inline public/index.html]: Rename application title to `Barefoot Nomad`

File: src/components/App.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Provider } from 'react-redux';
+import store from '../Redux/store/index';
+
+
+const App = ({ store }) => (
+  <Provider store={store}>
+
+  </Provider>
+)
+ReactDOM.render(<App store={store} />, document.getElementById('app'));
```
Comment (NiyoEric) [inline src/components/App.js]: I think you need to sync with @nziokaivy. She is working on setting up redux.

**Review by NiyoEric — CHANGES_REQUESTED** [review]
Good configuration @hezronkimutai. I have left some feedback below.

**Review by NiyongaboEric — APPROVED** [review]

**Review by NiyoEric — APPROVED** [review]

**Review by victkarangwa — APPROVED** [review]
Well done!

**Review by william000000 — APPROVED** [review]
Nice Job @hezronkimutai

**Review by nziokaivy — APPROVED** [review]
Looks good to me! Great Job @hezronkimutai

**Review by shemaeric — APPROVED** [review]
Great job @hezronkimutai

**Pull request merged**